# Uncovering the Key Dimensions of High-Throughput Biomolecular Data using Deep Learning
This is a deep learning framework, termed as DeepAE, to identify the key dimensions from high-dimensional gene expression profiles. DeepAE is composed of an input layer, seven hidden layers, and output layer to form the encoder and decoder phases that are corresponded to the compression and decompression. 

# PREREQUISITE
DeepAE was conducted by using Python 3 version and R v3.5.3. 
Following Python packages should be installed:
<ul>
<li><p>Tensorflow</p></li>
<li><p>numpy</p></li>
<li><p>pandas</p></li>
</ul>
Following R packages should be installed:
<ul>
<li><p>DOSE</p></li>
<li><p>topGO</p></li>
<li><p>RCy3</p></li>
<li><p>clusterProfiler</p></li>
<li><p>rWikiPathways</p></li>
<li><p>clusterHeatmap</p></li>  
<li><p>org.Mm.eg.db</p></li> 
<li><p>org.Hs.eg.db</p></li>  
<li><p>org.Sc.sgd.db</p></li> 
</ul>

# CONTAINS:
<ul>
<li><p>DeepAE.ipynb : Python script to run DeepAE and get results</p></li>
<li><p>WikiPathways import to Cytospace.R : R script to get the pathways and import to Cytospace App</p></li>
<li><p>Gene Ontology and WikiPathways analysis.R : R script to get the Gene Ontology and WikiPathways</p></li> 
<li><p>clusterHeatmap.R : R script to get the clustering heatmaps graphics</p></li> 
<li><p>Plot_Bars_ceytometry_data.py : Python script to plot the bar graphics for Ceytometry datasets</p></li>   
<li><p>Plot_Bars_metabolic_data.py : Python script script to plot the bar graphics for Metabolic datasets</p></li>    
<li><p>Plot_Bars_transcriptomic data.py : Python script to plot the bar graphics for transcriptomic datasets</p></li>     
<li><p>Find_high_weighted_genes_from_key_hidden_dimensions.py : Python script to selectthe high weighted genes from key hidden dimensions</p></li>
<li><p>Flt-SNE : Python scripts to get the t-SNE graphics</p></li> 
</ul>

---------------------------------------
Shixiong Zhang

sxzhang7-c@my.cityu.edu.hk

April 10 2019
